### ISSUE TITLE
False positives on generic pydantic models

### ISSUE THREAD

**Issue by GonozalIX** [pylint-dev/pylint#10602]
### Bug description

If one inherits from [generic pydantic models](https://docs.pydantic.dev/latest/concepts/models/#creating-models-without-validation) pylint ignores they are pydantic classes and throw a `too-few-public-methods` error (so implicit inheritance from pydantic `BaseModel` is ignored). 

```python
from typing import Generic, TypeVar
from pydantic import BaseModel

T = TypeVar("T")

class GenericModel(BaseModel, Generic[T]):
    prop1: T

class ChildModel(GenericModel[int]):
    prop2: int

print(ChildModel(prop1=1, prop2=2))
```

### Configuration

```ini

```

### Command used

```shell
pre-commit run pylint --files=playground.py
```

### Pylint output

```python
playground.py:9:0: R0903: Too few public methods (0/2) (too-few-public-methods)
```

### Expected behavior

It should be considered that `ChildModel` also inherits from `BaseModel` and the error should be suppressed… if you inherit from `BaseModel` explicitly everthing works as intended, also with no Generic Models chain inheritance works fine.

### Pylint version

```shell
pylint 3.3.8
```

### OS / Environment

Ubuntu

### Additional dependencies

```python
pydantic                  2.12.0a1 
pydantic-core             2.37.2
```
